### PR TITLE
x11-misc/yabar: new package

### PR DESCRIPTION
**Pull request by jakelyz**
A modern and lightweight status bar for X window managers.

Bug: https://bugs.gentoo.org/689866
Signed-off-by: Jake Lilly <lillyjsm@gmail.com>
Package-Manager: Portage-2.3.66, Repoman-2.3.11

### CODE REVIEW

File: x11-misc/yabar/Manifest [hunk 1]
```diff
@@ -0,0 +1 @@
+DIST yabar-a0d3fdfed992149b741eb8fcf53f02b5d1a6142e.tar.gz 892725 BLAKE2B eb14b06327533034756885c68ab03b1ff049c987d1dac6d083c553646cd178210429d9457c1067db3211fcb39e9814a260aed201dea4e6410544b9247c5dee93 SHA512 f93250605c5b56d64c7ec6a7e7faa6120cdc06191ddafb13822875fd98b1b7a1822d9b93b62c49c302b1d4bffb14a200feed937efb0b11711de7159ce83eb7c5
```
Comment (juippis) [inline x11-misc/yabar/Manifest]: This version isn't really recognizable. I see upstream has tagged releases, so use something like like `0.4.0_p20190720` in here instead please.

File: x11-misc/yabar/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+<pkgmetadata>
+	<maintainer type="person">
+		<email>lillyjsm@gmail.com</email>
+		<name>Jake Lilly</name>
+	</maintainer>
+	<maintainer type="project">
+		<email>proxy-maint@gentoo.org</email>
+		<name>Proxy Maintainers</name>
+	</maintainer>
+	<longdescription lang="en">Yabar is a modern and lightweight status bar that is intended to be used along with minimal X window managers.</longdescription>
```
Comment (juippis) [inline x11-misc/yabar/metadata.xml]: This is the same length as default DESCRIPTION in ebuild, and can be removed.

File: x11-misc/yabar/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	<use>
+		<flag name="playerctl">Uses <pkg>media-sound/playerctl</pkg> to retrieve the name of the currently played song.</flag>
```
Comment (juippis) [inline x11-misc/yabar/metadata.xml]: No dot in the end here please, to try and follow common style.

File: x11-misc/yabar/yabar-0.4.0_p20190316.ebuild [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+DEPEND="
+	x11-libs/cairo[xcb]
+	x11-libs/pango
+	x11-libs/xcb-util-wm
+	x11-libs/gdk-pixbuf
+	x11-libs/libxkbcommon[X]
+	media-libs/alsa-lib
+	net-wireless/wireless-tools
+	dev-libs/libconfig
+	app-text/asciidoc
```
Comment (juippis) [inline x11-misc/yabar/yabar-0.4.0_p20190316.ebuild]: Please sort the unconditional deps.
Could you also check all the slot operators and subslot bindings that are missing?

File: x11-misc/yabar/yabar-0.4.0_p20190316.ebuild [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		emake PLAYERCTL=1
+	else
+		emake
+	fi
+}
+
+src_install() {
+	default
+	docinto examples
+	dodoc "${S}"/examples/example.config
```
Comment (juippis) [inline x11-misc/yabar/yabar-0.4.0_p20190316.ebuild]: Just a minor thing, but `${S}` should be the CWD.

Comment (jakelyz) [reply x11-misc/yabar/yabar-0.4.0_p20190316.ebuild]: Fixed in newest push

File: x11-misc/yabar/yabar-0.4.0_p20190316.ebuild (hunk 4 above)
Comment (jakelyz) [reply x11-misc/yabar/yabar-0.4.0_p20190316.ebuild]: Re-ordered and slots considered in newest push

File: x11-misc/yabar/metadata.xml (hunk 3 above)
Comment (jakelyz) [reply x11-misc/yabar/metadata.xml]: removed

File: x11-misc/yabar/Manifest (hunk 1 above)
Comment (jakelyz) [reply x11-misc/yabar/Manifest]: changes made to reflect tagged release in manifest

**Comment by juippis** [comment]
```
 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/yabar
```

You should fix that. Otherwise looks good to me. Then squash your commits into 1.

File: x11-misc/yabar/yabar-0.4.0_p20190316.ebuild [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+src_prepare() {
+	default
+	# Respect CFLAGS and LDFLAGS
+	sed -i -e "s:-flto -O2::g" \
+	-e "s:-Wall::" Makefile || die "Failed to update Makefile"
+
+	# Declare VERSION without relying on git,
+	# and replace playerctl dependency option to base name
+	sed -i -e "s/\$(shell git describe)/0.4.0-179-ga0d3fdf/" \
```
Comment (juippis) [inline x11-misc/yabar/yabar-0.4.0_p20190316.ebuild]: (already mentioned this in IRC but wanna state it here for the record as well: these 2 sed commands can be combined into one, as sed takes multiple arguments when editing the same file)

Comment (jakelyz) [reply x11-misc/yabar/yabar-0.4.0_p20190316.ebuild]: Fixed

**Comment by jakelyz** [comment]
> please correct your signed-off-by:
> ` Signed-off-by: Random J Developer <random@developer.example.org>`
> 
> documentation:
> https://www.kernel.org/doc/Documentation/process/submitting-patches.rst

fixed commit msg

**Comment by juippis** [comment]
I cleaned the metadata.xml file a bit. 

For the next version, maybe look into some possibility of not pulling wireless_tools, since I dont have wireless network and running yabar just prints tons of errors about wifi status?

**Pull request closed without merging**